backend/src/v1/staging_complaint/staging_complaint.service.ts

```diff
@@ -53,9 +59,56 @@ export class StagingComplaintService {
     return newStagingComplaint;
   }
 
-  async process(complaintIdentifier: string): Promise<any> {
+  // Creates a Complaint Update staging record.  Used when WebEOC creates a complaint update.
+  async createComplaintUpdate(stagingComplaint: WebEOCComplaintUpdate): Promise<StagingComplaint> {
+    // ignore existing updates of the same incident number and update number, they already exist in the staging table
+    const existingStagingComplaint = await this.stagingComplaintRepository
+      .createQueryBuilder("stagingComplaint")
+      .leftJoinAndSelect("stagingComplaint.stagingActivityCode", "stagingActivityCode")
+      .where("stagingComplaint.complaintIdentifer = :complaintIdentifier", {
+        complaintIdentifier: stagingComplaint.parent_incident_number,
+      })
+      .andWhere("stagingActivityCode.stagingActivityCode = :activityCode", { activityCode: "UPDATE" })
+      .andWhere(
+        new Brackets((qb) => {
+          qb.where("stagingComplaint.complaint_jsonb @> :updateNumber", {
+            updateNumber: { update_number: stagingComplaint.update_number },
+          }).orWhere("stagingComplaint.complaint_jsonb IS NULL");
+        }),
+      )
+      .getOne();
+
+    // ignore duplicates
+    if (existingStagingComplaint) {
+      return;
+    }
+
+    const newStagingComplaint = this.stagingComplaintRepository.create();
+    newStagingComplaint.stagingStatusCode = { stagingStatusCode: StagingStatusCodeEnum.PENDING } as StagingStatusCode;
+    newStagingComplaint.stagingActivityCode = {
+      stagingActivityCode: StagingActivityCodeEnum.UPDATE,
+    } as StagingActivityCode;
+    newStagingComplaint.complaintIdentifer = stagingComplaint.parent_incident_number;
+    newStagingComplaint.complaintJsonb = stagingComplaint;
+    newStagingComplaint.createUserId = "WebEOC";
```

Comment:
minor suggestion here
```
const userId = "WebEOC";
const current = new Date();

newStagingComplaint.createUserId = userId
newStagingComplaint.createUtcTimestamp = current
newStagingComplaint.updateUserId = userId
newStagingComplaint.updateUtcTimestamp = current
```

rational for this is that if we need to change the user, there's only one place to do it vs two, and it keeps the create and update dates inline